app/controllers/spree/admin/products_controller_decorator.rb

```diff
@@ -31,6 +31,18 @@ def index
     @show_latest_import = params[:latest_import] || false
   end
 
+  def create
```

Comment:
I'm not a big fan of Rails' callbacks, but this seems the perfect use case for `around_action`:
```rb
around_action :delete_stock_params_and_set_after, only: :create, :update
```
https://guides.rubyonrails.org/action_controller_overview.html#after-filters-and-around-filters

Reply:
ehehe, I didnt now that existed, thanks! but tbh I think these callbacks are terrible. It would reduce lines of code but the current solution is very explicit about what's doing. When I am doing things that are not standard (it's the case here) I like to be explicit. around_action would hide it on the top of the file. I prefer as is...